### PR TITLE
subversion and libssh: add missing krb5 build dependency

### PR DESCRIPTION
**Pull request by maxim-belkin**
Standalone linux installation.
When building `subversion` and `libssh` from source, both complain about missing `gssapi/gssapi.h` file:

**Subversion:**
```
[linuxbrew-folder]/bin/gcc-5 -o auth/auth_spnego_gss.o -c -Os -w -pipe -march=core2 -std=c89 -Wdeclaration-after-statement -Wmissing-prototypes -Wall -O2 -pthread -DNDEBUG -DLINUX -D_REENTRANT -D_GNU_SOURCE -DSERF_HAVE_GSSAPI -I. -I[linuxbrew-folder]/Cellar/apr/1.5.2/libexec/include/apr-1 -I[linuxbrew-folder]/Cellar/apr-util/1.5.4_1/libexec/include/apr-1 -I[linuxbrew-folder]/opt/openssl/include auth/auth_spnego_gss.c
auth/auth_spnego_gss.c:22:27: fatal error: gssapi/gssapi.h: No such file or directory
compilation terminated.
scons: *** [auth/auth_spnego_gss.o] Error 1
scons: building terminated because of errors.
```

**Libssh:**
```
/tmp/libssh20160224-15635-9lh8wl/libssh-0.7.2/src/gssapi.c:30:27: fatal error: gssapi/gssapi.h: No such file or directory                                                                                                              
compilation terminated.                                                                                                                                                                                                                
make[2]: *** [src/CMakeFiles/ssh_shared.dir/gssapi.c.o] Error 1                                                                                                                                                                        
make[2]: *** Waiting for unfinished jobs....                                                                                                                                                                                           
make[2]: Leaving directory `/tmp/libssh20160224-15635-9lh8wl/libssh-0.7.2/build'                                                                                                                                                       
make[1]: *** [src/CMakeFiles/ssh_shared.dir/all] Error 2                                                                                                                                                                               
make[1]: Leaving directory `/tmp/libssh20160224-15635-9lh8wl/libssh-0.7.2/build'                                                                                                                                                       
make: *** [all] Error
```
(By the by, `libssh` has 0.7.3 now)

This file is in `/usr/include/gssapi/gssapi.h`, but standalone installation does not know about it.
Package `krb5` provides us with this file, hence the PR.

P.S. building `subversion` still fails but with a different error `/usr/bin/ld: unrecognized option '-plugin'`
Not sure why `/usr/bin/ld` is called...
```
[linuxbrew-folder]/bin/gcc-5 -o libserf-1.so.1.3.0 -L[linuxbrew-folder]/lib -Wl,--dynamic-linker=[linuxbrew-folder]/lib/ld.so -Wl,-rpath,[linuxbrew-folder]/lib -pthread -shared -Wl,-Bsymbolic -Wl,-soname=libserf-1.so.1 -Wl,-rpath=[linuxbrew-folder]/Cellar/subversion/1.9.3/lib -Wl,-rpath=:[linuxbrew-folder]/Cellar/apr/1.5.2/libexec/lib -Wl,-rpath=:[linuxbrew-folder]/Cellar/apr-util/1.5.4_1/libexec/lib -Wl,-rpath=:[linuxbrew-folder]/opt/openssl/lib context.os incoming.os outgoing.os ssltunnel.os buckets/aggregate_buckets.os buckets/allocator.os buckets/barrier_buckets.os buckets/buckets.os buckets/bwtp_buckets.os buckets/chunk_buckets.os buckets/dechunk_buckets.os buckets/deflate_buckets.os buckets/file_buckets.os buckets/headers_buckets.os buckets/iovec_buckets.os buckets/limit_buckets.os buckets/mmap_buckets.os buckets/request_buckets.os buckets/response_body_buckets.os buckets/response_buckets.os buckets/simple_buckets.os buckets/socket_buckets.os buckets/ssl_buckets.os auth/auth.os auth/auth_basic.os auth/auth_digest.os auth/auth_spnego.os auth/auth_spnego_gss.os auth/auth_spnego_sspi.os -L[linuxbrew-folder]/Cellar/apr/1.5.2/libexec/lib -L[linuxbrew-folder]/Cellar/apr-util/1.5.4_1/libexec/lib -L[linuxbrew-folder]/opt/openssl/lib -lssl -lcrypto -lz -lapr-1 -luuid -lrt -lcrypt -lpthread -ldl -laprutil-1 -lexpat -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err
/usr/bin/ld: unrecognized option '-plugin'
/usr/bin/ld: use the --help option for usage information
collect2: error: ld returned 1 exit status
scons: *** [libserf-1.so.1.3.0] Error 1
scons: building terminated because of errors.
```

### CODE REVIEW

**Comment by maxim-belkin** [comment]
I'm not exactly sure how PR work... when I click on the link to my patch ([8e5c254](https://github.com/Linuxbrew/linuxbrew/commit/8e5c2544046c00db76be57ee0fea0f8571a30588)) I see just the reported changes. When I go to [Files Changed](https://github.com/Linuxbrew/linuxbrew/pull/838/files) above, then I see 3 files changed (+`coreutils` formula where I change the url + add a `gperf` dependency... Not sure why this is happening...

**Comment by sjackman** [comment]
> (By the by, libssh has 0.7.3 now)

If you like, you may submit a pull request upstream to Homebrew/homebrew.

**Comment by sjackman** [comment]
> P.S. building subversion still fails but with a different error /usr/bin/ld: unrecognized option '-plugin'
Not sure why /usr/bin/ld is called...

Yes, that is odd. Do you have `binutils` installed by Linuxbrew? Report `which -a ld`

File: Library/Formula/coreutils.rb [hunk 1]
```diff
@@ -29,6 +29,7 @@ class Coreutils < Formula
     depends_on "texinfo" => :build
     depends_on "xz" => :build
     depends_on "wget" => :build
+    depends_on "homebrew/dupes/gperf" => :build unless OS.mac?
```
Comment (sjackman) [inline Library/Formula/coreutils.rb]: Please remove these changes as it's not relevant to this particular PR.

File: Library/Formula/libssh.rb [hunk 2]
```diff
@@ -14,6 +14,7 @@ class Libssh < Formula
 
   depends_on "cmake" => :build
   depends_on "openssl"
+  depends_on "krb5" => :build unless OS.mac?
```
Comment (sjackman) [inline Library/Formula/libssh.rb]: Please change this to `homebrew/dupes/krb5`. Is it only a build time dependency? The libraries aren't needed at run time?

**Comment by sjackman** [comment]
> I'm not exactly sure how PR work... when I click on the link to my patch (8e5c254) I see just the reported changes. When I go to Files Changed above, then I see 3 files changed (+coreutils formula where I change the url + add a gperf dependency... Not sure why this is happening...

It's best to create a branch for each particular change that you want to make. For example

```sh
git checkout origin/master
git checkout -b subversion
# Make your changes.
git push maxim-belkin subversion
# Create your pull request from the GitHub web interface.
git checkout master
```

**Comment by sjackman** [comment]
Please report `brew gist-logs subversion`

**Comment by sjackman** [comment]
As a workaround, try `brew install subversion --without-serf`
See #182

**Comment by sjackman** [comment]
Please also report `brew gist-logs libssh`

File: Library/Formula/libssh.rb (hunk 2 above)
Comment (maxim-belkin) [inline Library/Formula/libssh.rb]: No idea. But `gssapi.h` (and, therefore, `krb5`) is definitely necessary to build `libssh`.

**Comment by sjackman** [comment]
`krb5` is not a hard dependency of `libssh`. I ran
```sh
docker run -it sjackman/linuxbrew
sudo apt-get remove libkrb5-dev
brew install libssh
```
and it built okay. Giving the following message:
```
-- Could NOT find GSSAPI (missing:  GSSAPI_LIBRARIES GSSAPI_INCLUDE_DIR) 
-- GSSAPI support : 0
```
https://gist.github.com/aeaa7e1ebceaf8cff861

**Comment by sjackman** [comment]
`homebrew/dupes/krb5` seems to depend on `bison`
```
==> Installing krb5 from homebrew/dupes
…
yacc  getdate.y 
make[2]: yacc: Command not found
```

**Comment by maxim-belkin** [comment]
I don't see `yacc` as a dependency of `krb5`:
https://gist.github.com/anonymous/44aa6779dda952db5e0c

**Comment by maxim-belkin** [comment]
gist log for `libssh`: https://gist.github.com/6ea48ef886e3445540aa

**Comment by maxim-belkin** [comment]
I've noticed `doxygen` dependency in `libssh`. `Doxygen` depends on `bison`.

**Comment by sjackman** [comment]
Unrelated error when installing `curl`
https://travis-ci.org/Linuxbrew/linuxbrew/builds/111801055#L1229
```
/usr/bin/perl ./zsh.pl ../src/curl > _curl
/tmp/curl20160225-30127-6bwhx2/curl-7.47.1/src/.libs/lt-curl: error while loading shared libraries: liblber-2.4.so.2: cannot open shared object file: No such file or directory
```
`liblber` is related to LDAP. https://packages.debian.org/sid/amd64/libldap-2.4-2/filelist

**Comment by sjackman** [comment]
`bison` (which is a type of `yacc`) does seem to be a dependency of `krb5`:
https://gist.github.com/anonymous/44aa6779dda952db5e0c#file-01-configure-L386
https://gist.github.com/anonymous/44aa6779dda952db5e0c#file-02-make-L1800

**Comment by sjackman** [comment]
```
cd /tmp/libssh20160224-16876-1rdcxwv/libssh-0.7.3/build/src && /data/server4/mbelkin/.linuxbrew/bin/gcc-5  -DLIBSSH_EXPORTS -I/tmp/libssh20160224-16876-1rdcxwv/libssh-0.7.3/build/src -I/tmp/libssh20160224-16876-1rdcxwv/libssh-0.7.3/src -I/tmp/libssh20160224-16876-1rdcxwv/libssh-0.7.3/include -I/tmp/libssh20160224-16876-1rdcxwv/libssh-0.7.3/build -I/data/server4/mbelkin/.linuxbrew/include  -Os -w -pipe -march=core2  -std=gnu99 -pedantic -pedantic-errors -Wall -Wextra -Wshadow -Wmissing-prototypes -Wdeclaration-after-statement -Wunused -Wfloat-equal -Wpointer-arith -Wwrite-strings -Wformat-security -Wmissing-format-attribute -fPIC -fstack-protector -D_LARGEFILE64_SOURCE -DNDEBUG -fPIC   -fvisibility=hidden -o CMakeFiles/ssh_shared.dir/external/curve25519_ref.c.o   -c /tmp/libssh20160224-16876-1rdcxwv/libssh-0.7.3/src/external/curve25519_ref.c
/tmp/libssh20160224-16876-1rdcxwv/libssh-0.7.3/src/gssapi.c:30:27: fatal error: gssapi/gssapi.h: No such file or directory
compilation terminated.
make[2]: *** [src/CMakeFiles/ssh_shared.dir/gssapi.c.o] Error 1
```
https://gist.github.com/anonymous/6ea48ef886e3445540aa#file-02-make-L120-L123

**Comment by sjackman** [comment]
https://gist.github.com/anonymous/6ea48ef886e3445540aa#file-01-cmake-L37
```
-- Found GSSAPI: /usr/lib64/libgssapi_krb5.so;/usr/lib64/libkrb5.so;/usr/lib64/libk5crypto.so;/usr/lib64/libcom_err.so  
```
https://gist.github.com/anonymous/6ea48ef886e3445540aa#file-01-cmake-L147
```
-- GSSAPI support : ON
```

**Comment by sjackman** [comment]
It's picking up GSSAPI from your host system. `brew install gcc --with-glibc` should fix that.
See https://github.com/Linuxbrew/linuxbrew/wiki/Standalone-Installation

**Comment by sjackman** [comment]
Your subversion issue might be fixed by #844.

**Comment by maxim-belkin** [comment]
https://gist.github.com/4062531b37355aa51d30

**Comment by sjackman** [comment]
`/usr/bin/ld: unrecognized option '-plugin'`
https://gist.github.com/anonymous/4062531b37355aa51d30#file-01-scons-L77

Do you have binutils installed by Linuxbrew? Report `which -a ld`
This issue is probably resolved by #845.

**Comment by sjackman** [comment]
Please submit one pull request per formula.

**Pull request closed without merging**